### PR TITLE
Make engine compile as static library, fix linux compiling and fix `renderer.cpp` compiler warnings

### PR DESCRIPTION
**Pull request by retcinder**
These changes have been tested on Arch Linux. May need further modification for other OSes.

### CODE REVIEW

**Comment by tulpenkiste** [comment]
Forgot to mention, this fixes issue #1

File: src/renderer.cpp [hunk 1]
```diff
@@ -88,11 +88,6 @@ int Renderer::Init(void (&Start)() ,void (&Update)(float deltaTime))
     glDeleteShader(vertexShader);
     glDeleteShader(fragmentShader);
 
-    // Vertex buffer
-    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float),
-                            (void *)0);
-    glEnableVertexAttribArray(0);
```
Comment (Ceebox) [inline src/renderer.cpp:94]: Good change, tested this with this change and everything works as normal.

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+CXX_FLAGS = -O3 -Wall -Wno-missing-braces -I ./include/glad/ -I ./include/
+
+ifeq ($(OS),Windows_NT)
+	# Windows specific compile flags
+	CXX_FLAGS += -I ./include/KHR/ -I ./include/GLFW/ -lopengl32 -lgdi32 -lwinmm -L ./lib/
+else
+	CXX_FLAGS += -lGL -lX11 -ldl
+endif
+
+CXX_FLAGS += -lglfw -lglad
```
[the remark this answers is not in the document]
Comment (tulpenkiste) [reply Makefile:13]: Ah yeah right, windows has a different library to link against.
Should be a simple fix.

[the remark this answers is not in the document]
Comment (retcinder) [reply Makefile]: @Ceebox I've managed to successfully compile it for Windows using the most recent changes I've made.

**Pull request closed without merging**